In [97]:
import pandas as pd
import numpy as np

In [98]:
Food = pd.read_csv("C:/Users/user/Documents/python/Dacon/data/식수예측/4_최종병합본/0321_기상청_코로나_식수(train data)_병합본.csv", encoding = "cp949")
Food.head(3)

일자  평균기온(℃)  강수량(mm)  일조합  defcnt 요일  본사정원수  본사휴가자수  본사출장자수  \
0  2016-02-01     -0.6      0.0  9.1       0  월   2601      50     150   
1  2016-02-02     -2.3      0.0  9.5       0  화   2601      50     173   
2  2016-02-03     -1.7      0.0  9.5       0  수   2601      56     180   

   본사시간외근무명령서승인건수  ...  new_col(중식) new_col(석식)    중식_1   중식_2   중식_3  \
0             238  ...            x           x  쌀밥/잡곡밥  오징어찌개   쇠불고기   
1             319  ...            x           x  쌀밥/잡곡밥   김치찌개  가자미튀김   
2             111  ...            x           x    카레덮밥   팽이장국   치킨핑거   

       석식_1   석식_2     석식_3 중식_1_CLEAN 석식_1_CLEAN  
0    쌀밥/잡곡밥    육개장  자반고등어구이       일반 밥       일반 밥  
1  콩나물밥*양념장    어묵국      유산슬       일반 밥       콩나물밥  
2    쌀밥/잡곡밥  청국장찌개   황태양념구이       카레덮밥       일반 밥  

[3 rows x 26 columns]

### 범주형 처리

In [99]:
Food = Food[["일자","평균기온(℃)","강수량(mm)","일조합", "defcnt", "요일","본사정원수","본사휴가자수","본사출장자수","본사시간외근무명령서승인건수","현본사소속재택근무자수", "중식계", "석식계", "new_col(중식)","new_col(석식)","중식_2","중식_3","석식_2","석식_3","중식_1_CLEAN","석식_1_CLEAN"]]
Food.tail(3)

일자  평균기온(℃)  강수량(mm)  일조합  defcnt 요일  본사정원수  본사휴가자수  본사출장자수  \
1202  2021-01-22      7.0      1.2  0.0   23092  금   2983     255     248   
1203  2021-01-25      6.3      0.0  9.0   23453  월   2983     107     153   
1204  2021-01-26      4.7     10.3  0.0   23557  화   2983      69     183   

      본사시간외근무명령서승인건수  ...   중식계  석식계  new_col(중식) new_col(석식)       중식_2  \
1202               1  ...   579  217            x           x       계란파국   
1203             616  ...  1145  502            x           x  전주식콩나물해장국   
1204             551  ...  1015  480            x           o      들깨미역국   

         중식_3   석식_2         석식_3 중식_1_CLEAN 석식_1_CLEAN  
1202      요거닭   수제비국     수제맛쵸킹탕수육      낙지비빔밥        흑미밥  
1203  돈육간장불고기  열무된장국  장어강정*데리야끼소스       일반 밥        흑미밥  
1204   교촌간장치킨   잔치국수       수제고기육전       일반 밥  할라피뇨멸치주먹밥  

[3 rows x 21 columns]

In [100]:
# 범주형을 수치형을 바꾸기 1 \
Food['일자']=Food['일자'].factorize()[0]
Food['요일']=Food['요일'].factorize()[0]
Food['new_col(중식)']=Food['new_col(중식)'].factorize()[0]
Food['new_col(석식)']=Food['new_col(석식)'].factorize()[0]
Food['중식_2']=Food['중식_2'].factorize()[0]
Food['중식_3']=Food['중식_3'].factorize()[0]
Food['석식_2']=Food['석식_2'].factorize()[0]
Food['석식_3']=Food['석식_3'].factorize()[0]
Food['중식_1_CLEAN']=Food['중식_1_CLEAN'].factorize()[0]
Food['석식_1_CLEAN']=Food['석식_1_CLEAN'].factorize()[0]

Food.isnull().sum()

일자                0
평균기온(℃)           0
강수량(mm)           0
일조합               0
defcnt            0
요일                0
본사정원수             0
본사휴가자수            0
본사출장자수            0
본사시간외근무명령서승인건수    0
현본사소속재택근무자수       0
중식계               0
석식계               0
new_col(중식)       0
new_col(석식)       0
중식_2              0
중식_3              0
석식_2              0
석식_3              0
중식_1_CLEAN        0
석식_1_CLEAN        0
dtype: int64

### CORR

In [101]:
import matplotlib.pyplot as plt
import seaborn as sns

In [102]:
Food.corr()

일자   평균기온(℃)   강수량(mm)       일조합    defcnt        요일  \
일자              1.000000 -0.034949  0.025594  0.018220  0.442684 -0.009087   
평균기온(℃)        -0.034949  1.000000  0.161477 -0.041889 -0.183105 -0.004688   
강수량(mm)         0.025594  0.161477  1.000000 -0.428070 -0.029233 -0.013527   
일조합             0.018220 -0.041889 -0.428070  1.000000 -0.026453  0.035585   
defcnt          0.442684 -0.183105 -0.029233 -0.026453  1.000000 -0.016920   
요일             -0.009087 -0.004688 -0.013527  0.035585 -0.016920  1.000000   
본사정원수           0.811343  0.067854  0.034379 -0.025194  0.317234 -0.008577   
본사휴가자수          0.169125  0.206580  0.022678  0.038340  0.052198  0.155578   
본사출장자수         -0.063490  0.118252 -0.024423 -0.003446 -0.223058  0.532962   
본사시간외근무명령서승인건수  0.182459 -0.052815 -0.007575  0.022612  0.095518 -0.554359   
현본사소속재택근무자수     0.574125 -0.024972  0.015028  0.041483  0.651355 -0.026795   
중식계            -0.112966 -0.093201  0.088758 -0.087874  0.078179 -0.731563   
석식계            -0.224198  0.008740  0.027613  0.018359 -0.099508 -0.312112   
new_col(중식)     0.235357 -0.058029 -0.005364 -0.044167  0.345897 -0.000348   
new_col(석식)     0.212942 -0.082479 -0.035132  0.017310  0.377934 -0.013760   
중식_2            0.472644 -0.001127  0.008731  0.041250  0.414384  0.043800   
중식_3            0.569223 -0.002386  0.035379 -0.025689  0.501326  0.049187   
석식_2            0.437837  0.007182  0.043051  0.006947  0.369547  0.010430   
석식_3            0.651655 -0.030223  0.023236  0.038482  0.512104 -0.027928   
중식_1_CLEAN      0.115527  0.011355  0.002922  0.008291 -0.012500  0.099975   
석식_1_CLEAN      0.472119  0.008935  0.030715 -0.038417  0.604833 -0.005786   

                   본사정원수    본사휴가자수    본사출장자수  본사시간외근무명령서승인건수  ...       중식계  \
일자              0.811343  0.169125 -0.063490        0.182459  ... -0.112966   
평균기온(℃)         0.067854  0.206580  0.118252       -0.052815  ... -0.093201   
강수량(mm)         0.034379  0.022678 -0.024423       -0.007575  ...  0.088758   
일조합            -0.025194  0.038340 -0.003446        0.022612  ... -0.087874   
defcnt          0.317234  0.052198 -0.223058        0.095518  ...  0.078179   
요일             -0.008577  0.155578  0.532962       -0.554359  ... -0.731563   
본사정원수           1.000000  0.244666 -0.061688        0.146544  ... -0.115529   
본사휴가자수          0.244666  1.000000 -0.273318       -0.230789  ... -0.391975   
본사출장자수         -0.061688 -0.273318  1.000000       -0.308757  ... -0.512680   
본사시간외근무명령서승인건수  0.146544 -0.230789 -0.308757        1.000000  ...  0.535611   
현본사소속재택근무자수     0.452293  0.065397 -0.352643        0.156792  ...  0.076509   
중식계            -0.115529 -0.391975 -0.512680        0.535611  ...  1.000000   
석식계            -0.173852 -0.316894 -0.188164        0.571168  ...  0.508287   
new_col(중식)     0.187562  0.038449 -0.071760        0.030815  ...  0.036582   
new_col(석식)     0.165326  0.017416 -0.092545        0.137034  ...  0.049270   
중식_2            0.352714  0.088756 -0.075516        0.105016  ... -0.051391   
중식_3            0.442811  0.114714 -0.091787        0.044042  ... -0.057068   
석식_2            0.340891  0.153378 -0.124412        0.087763  ... -0.027550   
석식_3            0.497473  0.115351 -0.156750        0.171385  ...  0.004479   
중식_1_CLEAN      0.092100  0.008906  0.072310       -0.297785  ... -0.096516   
석식_1_CLEAN      0.391969  0.083718 -0.149246        0.255867  ... -0.005073   

                     석식계  new_col(중식)  new_col(석식)      중식_2      중식_3  \
일자             -0.224198     0.235357     0.212942  0.472644  0.569223   
평균기온(℃)         0.008740    -0.058029    -0.082479 -0.001127 -0.002386   
강수량(mm)         0.027613    -0.005364    -0.035132  0.008731  0.035379   
일조합             0.018359    -0.044167     0.017310  0.041250 -0.025689   
defcnt         -0.099508     0.345897     0.377934  0.414384  0.501326   
요일             -0.312112    -0.000348    -0.013760  0.043800  0.049187   
본사정원수          -0.173852     0.187562     0.1653

-- corr --
- 현본사소속재택근무자수/defcnt => 0.65
- 중식_3/defcnt => 0.50
- 석식_3/defcnt => 0.51
- 석식_1_CLEAN/dffcnt => 0.60
- 본사출장자수/요일 => 0.53
- 본사시간외근무명령서승인건수/요일 => -0.55
- 중식계/요일 => -0.73
- 중식계/본사출장자수 => -0.51
- 중식계/본사시간외근무명령서승인건수 => 0.53
- 석식계/본사시간외근무명령서승인건수 => 0.57
- 등

### 회귀분석
- ols 함수로 회귀분석을 실시
- 종속변수 ~ 독립변수의 형태로 모형식

#### 전체 회귀분석 (석식계)

In [103]:
from statsmodels.formula.api import ols
import statsmodels.api as sm

In [104]:
# 종속 / 독립변수 
feature_columns = list(Food.columns.difference(["중식계", "석식계","일자"]))

X_data = Food[feature_columns]  # 독립변수
X_data.head()

y_data = Food["석식계"]  # 종속변수

In [105]:
# 독립변수에 상수항을 고려
x_data1 = sm.add_constant(X_data, has_constant = "add") 

C:\Users\user\anaconda3\lib\site-packages\statsmodels\tsa\tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


const  defcnt  new_col(석식)  new_col(중식)  강수량(mm)  본사시간외근무명령서승인건수  본사정원수  \
0       1.0       0            0            0      0.0             238   2601   
1       1.0       0            0            0      0.0             319   2601   
2       1.0       0            0            0      0.0             111   2601   
3       1.0       0            0            0      0.0             355   2601   
4       1.0       0            0            0      0.0              34   2601   
...     ...     ...          ...          ...      ...             ...    ...   
1200    1.0   22852            0            0      0.0               4   2983   
1201    1.0   22977            0            0      1.7             462   2983   
1202    1.0   23092            0            0      1.2               1   2983   
1203    1.0   23453            0            0      0.0             616   2983   
1204    1.0   23557            1            0     10.3             551   2983   

      본사출장자수  본사휴가자수  석식_1_CLEAN  석식_2  석식_3  요일  일조합  중식_1_CLEAN  중식_2  중식_3  \
0        150      50           0     0     0   0  9.1           0     0     0   
1        173      50           1     1     1   1  9.5           0     1     1   
2        180      56           0     2     2   2  9.5           1     2     2   
3        220     104           2     3     3   3  6.3           0     3     3   
4        181     278           0     4     4   4  9.6           0     4     4   
...      ...     ...         ...   ...   ...  ..  ...         ...   ...   ...   
1200     198      75          13   284   437   2  8.9           0    95   385   
1201     231      92          77    34   438   3  5.0           0     8   135   
1202     248     255          77    27   439   4  0.0           2   267   362   
1203     153     107          77    13   440   0  9.0           0   268   128   
1204     183      69          97    14   441   1  0.0           0   113   386   

      평균기온(℃)  현본사소속재택근무자수  
0        -0.6            0  
1        -2.3            0  
2        -1.7            0  
3        -0.2            0  
4         1.3            0  
...       ...          ...  
1200     -0.4          391  
1201      5.2          351  
1202      7.0          303  
1203      6.3          327  
1204      4.7          362  

[1205 rows x 19 columns]

In [106]:
# OLS : 잔차제곱의합을 최소화하는 것
multi_model = sm.OLS(y_data, x_data1)
fitted_multi_model = multi_model.fit()

In [107]:
fitted_multi_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    석식계   R-squared:                       0.462
Model:                            OLS   Adj. R-squared:                  0.454
Method:                 Least Squares   F-statistic:                     56.51
Date:                Fri, 25 Mar 2022   Prob (F-statistic):          2.72e-145
Time:                        14:40:38   Log-Likelihood:                -7283.8
No. Observations:                1205   AIC:                         1.461e+04
Df Residuals:                    1186   BIC:                         1.470e+04
Df Model:                          18                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const            828.9916     59.526     13.927      0.000     712.204     945.779
defcnt            -0.0013      0.001     -0.913      0.362      -0.004       0.002
new_col(석식)       -3.5906     25.121     -0.143      0.886     -52.878      45.696
new_col(중식)       32.1121     22.579      1.422      0.155     -12.186      76.411
강수량(mm)            0.3572      0.257      1.389      0.165      -0.147       0.862
본사시간외근무명령서승인건수     0.3218      0.017     19.122      0.000       0.289       0.355
본사정원수             -0.0992      0.022     -4.409      0.000      -0.143      -0.055
본사출장자수            -0.7547      0.108     -6.980      0.000      -0.967      -0.543
본사휴가자수            -0.2528      0.027     -9.287      0.000      -0.306      -0.199
석식_1_CLEAN        -0.1376      0.174     -0.793      0.428      -0.478       0.203
석식_2               0.1765      0.050      3.513      0.000       0.078       0.275
석식_3              -0.0729      0.037     -1.977      0.048      -0.145      -0.001
요일                17.0250      3.165      5.379      0.000      10.816      23.234
일조합                0.9825      0.854      1.151      0.250      -0.693       2.658
중식_1_CLEAN        -1.2454      0.555     -2.245      0.025      -2.334      -0.157
중식_2               0.0029      0.058      0.050      0.960      -0.111       0.117
중식_3              -0.1199      0.040     -2.983      0.003      -0.199      -0.041
평균기온(℃)            1.7597      0.356      4.948      0.000       1.062       2.457
현본사소속재택근무자수       -0.1023      0.048     -2.122      0.034      -0.197      -0.008
==============================================================================
Omnibus:                      403.380   Durbin-Watson:                   1.858
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1575.853
Skew:                          -1.574   Prob(JB):                         0.00
Kurtosis:                       7.635   Cond. No.                     7.15e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 7.15e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

- R^2 : 44.8% -> 46.2%
- 유의하지 않은 변수(P-value > 0.05) : defcnt, new_col(석식), new_col(중식), 강수량(mm), 석식_1_CLEAN, 일조합, 중식_2

#### 전체 회귀분석 (중식계)

In [114]:
# 종속 / 독립변수 
feature_columns = list(Food.columns.difference(["중식계", "석식계","일자"]))

X_data = Food[feature_columns]  # 독립변수
X_data.head()

y_data = Food["중식계"]  # 종속변수

In [115]:
# 독립변수에 상수항을 고려
x_data1 = sm.add_constant(X_data, has_constant = "add") 

C:\Users\user\anaconda3\lib\site-packages\statsmodels\tsa\tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [116]:
# OLS : 잔차제곱의합을 최소화하는 것
multi_model = sm.OLS(y_data, x_data1)
fitted_multi_model = multi_model.fit()

In [117]:
fitted_multi_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    중식계   R-squared:                       0.731
Model:                            OLS   Adj. R-squared:                  0.726
Method:                 Least Squares   F-statistic:                     178.7
Date:                Fri, 25 Mar 2022   Prob (F-statistic):          5.78e-322
Time:                        14:49:58   Log-Likelihood:                -7359.6
No. Observations:                1205   AIC:                         1.476e+04
Df Residuals:                    1186   BIC:                         1.485e+04
Df Model:                          18                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const           1680.0840     63.390     26.504      0.000    1555.715    1804.453
defcnt             0.0068      0.002      4.480      0.000       0.004       0.010
new_col(석식)       30.6363     26.752      1.145      0.252     -21.850      83.123
new_col(중식)       77.6408     24.044      3.229      0.001      30.467     124.815
강수량(mm)            1.1708      0.274      4.275      0.000       0.634       1.708
본사시간외근무명령서승인건수     0.1290      0.018      7.196      0.000       0.094       0.164
본사정원수             -0.0545      0.024     -2.273      0.023      -0.101      -0.007
본사출장자수            -1.9723      0.115    -17.129      0.000      -2.198      -1.746
본사휴가자수            -0.5955      0.029    -20.541      0.000      -0.652      -0.539
석식_1_CLEAN        -0.8110      0.185     -4.388      0.000      -1.174      -0.448
석식_2               0.0430      0.054      0.803      0.422      -0.062       0.148
석식_3               0.0080      0.039      0.205      0.838      -0.069       0.085
요일               -54.1970      3.370    -16.081      0.000     -60.809     -47.585
일조합               -1.4102      0.909     -1.551      0.121      -3.194       0.374
중식_1_CLEAN         0.5392      0.591      0.913      0.362      -0.620       1.698
중식_2              -0.0265      0.062     -0.427      0.669      -0.148       0.095
중식_3               0.0104      0.043      0.242      0.809      -0.074       0.094
평균기온(℃)            1.3130      0.379      3.467      0.001       0.570       2.056
현본사소속재택근무자수       -0.1873      0.051     -3.648      0.000      -0.288      -0.087
==============================================================================
Omnibus:                       17.989   Durbin-Watson:                   1.542
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               31.492
Skew:                           0.010   Prob(JB):                     1.45e-07
Kurtosis:                       3.792   Cond. No.                     7.15e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 7.15e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

- R^2 : 71.9% -> 73.1%
- 유의하지 않은 변수 : new_col(석식), 석식_2, 석식_3, 일조합, 중식_1_CLEAN, 중식_2, 중식_3

#### 유의하지 않은 변수 제거(석식계)

In [120]:
# 종속 / 독립변수 
feature_columns = list(Food.columns.difference(["중식계", "석식계","일자", "defcnt", "new_col(석식)", "new_col(중식)", "강수량(mm)", "석식_1_CLEAN", "일조합", "중식_2"]))

X_data = Food[feature_columns]  # 독립변수
X_data.head()

y_data = Food["석식계"]  # 종속변수

In [121]:
# 독립변수에 상수항을 고려
x_data1 = sm.add_constant(X_data, has_constant = "add") 

# OLS : 잔차제곱의합을 최소화하는 것
multi_model = sm.OLS(y_data, x_data1)
fitted_multi_model = multi_model.fit()

fitted_multi_model.summary()

C:\Users\user\anaconda3\lib\site-packages\statsmodels\tsa\tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    석식계   R-squared:                       0.459
Model:                            OLS   Adj. R-squared:                  0.454
Method:                 Least Squares   F-statistic:                     91.97
Date:                Fri, 25 Mar 2022   Prob (F-statistic):          1.49e-150
Time:                        14:53:57   Log-Likelihood:                -7286.9
No. Observations:                1205   AIC:                         1.460e+04
Df Residuals:                    1193   BIC:                         1.466e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const            840.3267     58.348     14.402      0.000     725.850     954.804
본사시간외근무명령서승인건수     0.3200      0.016     19.761      0.000       0.288       0.352
본사정원수             -0.1001      0.022     -4.483      0.000      -0.144      -0.056
본사출장자수            -0.7587      0.107     -7.086      0.000      -0.969      -0.549
본사휴가자수            -0.2531      0.027     -9.372      0.000      -0.306      -0.200
석식_2               0.1722      0.050      3.457      0.001       0.074       0.270
석식_3              -0.0795      0.036     -2.227      0.026      -0.150      -0.009
요일                16.9733      3.131      5.420      0.000      10.830      23.117
중식_1_CLEAN        -1.1105      0.545     -2.036      0.042      -2.181      -0.040
중식_3              -0.1276      0.039     -3.273      0.001      -0.204      -0.051
평균기온(℃)            1.8700      0.339      5.516      0.000       1.205       2.535
현본사소속재택근무자수       -0.1196      0.042     -2.815      0.005      -0.203      -0.036
==============================================================================
Omnibus:                      394.025   Durbin-Watson:                   1.859
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1547.821
Skew:                          -1.531   Prob(JB):                         0.00
Kurtosis:                       7.631   Cond. No.                     5.61e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.61e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

- R^2는 44%로 크게 높아지지 않음 -> (기상청 추가한 후) 45.9%
- 모든 변수 유의

#### 유의하지 않은 변수 제거(중식계)

In [122]:
# 종속 / 독립변수 
feature_columns = list(Food.columns.difference(["중식계", "석식계","일자","new_col(석식)", "석식_2", "석식_3", "일조합", "중식_1_CLEAN", "중식_2", "중식_3"]))

X_data = Food[feature_columns]  # 독립변수
X_data.head()

y_data = Food["중식계"]  # 종속변수

In [123]:
# 독립변수에 상수항을 고려
x_data1 = sm.add_constant(X_data, has_constant = "add") 

# OLS : 잔차제곱의합을 최소화하는 것
multi_model = sm.OLS(y_data, x_data1)
fitted_multi_model = multi_model.fit()

fitted_multi_model.summary()

C:\Users\user\anaconda3\lib\site-packages\statsmodels\tsa\tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    중식계   R-squared:                       0.729
Model:                            OLS   Adj. R-squared:                  0.727
Method:                 Least Squares   F-statistic:                     292.2
Date:                Fri, 25 Mar 2022   Prob (F-statistic):               0.00
Time:                        14:55:09   Log-Likelihood:                -7362.4
No. Observations:                1205   AIC:                         1.475e+04
Df Residuals:                    1193   BIC:                         1.481e+04
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const           1651.7564     60.201     27.438      0.000    1533.646    1769.867
defcnt             0.0074      0.001      5.021      0.000       0.005       0.010
new_col(중식)       79.0354     23.840      3.315      0.001      32.263     125.808
강수량(mm)            1.3562      0.247      5.497      0.000       0.872       1.840
본사시간외근무명령서승인건수     0.1240      0.017      7.325      0.000       0.091       0.157
본사정원수             -0.0462      0.023     -2.031      0.042      -0.091      -0.002
본사출장자수            -1.9667      0.113    -17.329      0.000      -2.189      -1.744
본사휴가자수            -0.5972      0.028    -20.985      0.000      -0.653      -0.541
석식_1_CLEAN        -0.7876      0.178     -4.432      0.000      -1.136      -0.439
요일               -54.6300      3.353    -16.295      0.000     -61.208     -48.052
평균기온(℃)            1.2876      0.378      3.408      0.001       0.546       2.029
현본사소속재택근무자수       -0.1800      0.046     -3.906      0.000      -0.270      -0.090
==============================================================================
Omnibus:                       15.792   Durbin-Watson:                   1.554
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               26.374
Skew:                          -0.001   Prob(JB):                     1.87e-06
Kurtosis:                       3.725   Cond. No.                     6.78e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 6.78e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

- 모든 변수 유의함
- R^2은 73 -> 72%로 감소